mpxe/harness/pm.py

```diff
@@ -52,12 +51,12 @@ def stop_all(self):
     def poll(self):
         def prep_poll():
             poll = select.poll()
-            for fd in self.fd_to_proj.keys():
+            for fd in self.fd_to_proj:
                 poll.register(fd, select.POLLIN)
             return poll
 
         def handle_poll_res(fd, event):
-            if not (event & select.POLLIN):
+            if not event & select.POLLIN:
```

Comment:
IMO this change leaves the line ambiguous; I'd have to look up the Python precedence chart to make sure it's `not (event & select.POLLIN)` and not `(not event) & select.POLLIN`. Perhaps change it to `if (event & select.POLLIN) == 0:` - that version is unambiguous and accepted by pylint.